src/app_service/service/chat/service.nim

```diff
@@ -94,7 +95,19 @@ QtObject:
     result.contactService = contactService
     result.chats = initTable[string, ChatDto]()
 
+  # Forward declarations
+  proc updateOrAddChat*(self: Service, chat: ChatDto)
+  
   proc init*(self: Service) =
+    self.events.on(SignalType.Message.event) do(e: Args):
```

Comment:
@borismelnik just for the readability sake please add this to some private method like `doConnect`
```
proc doConnect(self: Service) =
    self.events.on(SignalType.Message.event) do(e: Args):
      var receivedData = MessageSignal(e)

      # Handling chat updates
      if (receivedData.chats.len > 0):
        for chatDto in receivedData.chats:
          self.updateOrAddChat(chatDto)
        self.events.emit(SIGNAL_CHAT_UPDATE, ChatUpdateArgsNew(messages: receivedData.messages, chats: receivedData.chats))
```